### PR TITLE
Absolute URLs in response's relations required? issues/#6

### PR DESCRIPTION
**Pull request by nln42**
an proposal for my issue #6

### CODE REVIEW

**Comment by nln42** [comment]
Sorry, I'm not sure why the unit test `com.mercateo.rest.hateoas.client.impl.sse.EventSourceWithCloseGuard0Test.testErrorClosing()`
fails in Travis CI  build? Local maven build works just fine... Any ideas?

File: src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseImpl.java [hunk 1]
```diff
@@ -20,6 +22,9 @@
 
     protected final T value;
 
+    @NonNull
+    private URI previousUri;
```
Comment (JoergAdler) [inline src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseImpl.java]: could you rename this to "uri" or something like that? As i get it right, this is the uri where the actual response is from?

File: src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseBuilder.java [hunk 2]
```diff
@@ -27,10 +28,11 @@
 	private Client client;
 	@NonNull
 	private ObjectMapper objectMapper;
+	URI uri = URI.create("http://localhost:8080/");
```
Comment (JoergAdler) [inline src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseBuilder.java]: actually, the uri is no state of the responsebuilder, it should be a parameter of the the "build" - methods.
also i don't know, what the local host adress means here :-)

Comment (JoergAdler) [inline src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseBuilder.java]: also, the uri is never changed, so the host will always be the inital one

Comment (nln42) [inline src/main/java/com/mercateo/rest/hateoas/client/impl/ResponseBuilder.java]: > As I see it right, this pull request only works for relative uris on the same host at the moment.

I don't understand this point. What do you mean with "only works for relative URIs on the same host"? I would say : of course, therefore they are relative... :-)

The solution is based on the behavior of `URI#resolve` method.

```
       URI.create("http://foo.bar/api")
                .resolve("/api/doc/")
                .equals(
                        URI.create("http://foo.bar/api/doc/"));

        URI.create("http://foo.bar/api") 
                .resolve("/api2/doc/") 
                .equals( 
                        URI.create("http://foo.bar/api2/doc/"));

        URI.create("http://foo.bar/api") //
                .resolve("http://mercateo.com/foo/api/") //
                .equals(
                        URI.create("http://mercateo.com/foo/api/"));
```

**Comment by JoergAdler** [comment]
could you pleas fix the compile errors in OngoingResponseImpl0Test.java?

**Comment by nln42** [comment]
Thanks a lot 👍

**Pull request merged**